README/ReleaseNotes/v628/index.md

```diff
@@ -76,6 +76,19 @@ $ rootreadspeed --files <local-folder>/File1.root xrootd://<url-folder>/File2.ro
 
 ## I/O Libraries
 
+- A new cross-protocol redirection has been added to allow files on EOS mounts to be
+  opened by `TFile::Open` via XRootD protocol rather than via FUSE when that is possible.
+  The redirection uses the `eos.url.xroot` extended file attribute that is present on files
+  in EOS. The attribute can be viewed with `getfattr -n eos.url.xroot [file]` on the command
+  line. When the URL passed into `TFile::Open` is a plain file name for a file on an EOS mount,
+  the extended attribute is used to attempt the redirection to XRootD protocol. If the URL
+  starts with `file://`, however, then the redirection is avoided, as in this case the call to
+  check the extended attribute will not be a valid path for a file and will fail with file not
+  found. Note that when ROOT files are opened via `TChain`, the `file://` prefix gets stripped
+  from the URL prior to any call to `TFile::Open`, therefore the redirection will always be
+  attempted in this case. If the redirection fails, the file is opened using the plain file path
+  as before. Note also that this feature is controlled by the pre-existing configuration option
+  `TFile.CrossProtocolRedirects` in `rootrc` and is therefore enabled by default.
```

Comment:
Here is an attempt to explain a bit more to the user why they want this feature:
```suggestion
- Faster reading from EOS.
  A new cross-protocol redirection has been added to `TFile::Open` to allow files on EOS
  mounts to be opened via the XRootD protocol. This considerably improves read speeds.
  The redirection uses the `eos.url.xroot` extended file attribute that is present on files
  in EOS. The attribute can be viewed with `getfattr -n eos.url.xroot [file]` on the command
  line. When the URL passed into `TFile::Open` is a plain file name for a file on an EOS mount,
  the extended attribute is used to attempt the redirection to the XRootD protocol. If the URL
  starts with `file://`, the redirection is disabled. Note that when ROOT files are opened via
  `TChain`, the `file://` prefix gets stripped from the URL automatically. Therefore, the
  redirection will always be attempted in a `TChain`.
  If a redirection fails, the file is opened using the plain file path as before. All redirections
  can be disabled setting `TFile.CrossProtocolRedirects` to 0 in `rootrc`.
```

Reply:
I took some of the suggestions in the end, as I had to touch the code and go through the CI again anyway.